File: buffer/gpu/src/jvmMain/java/com/wsr/gpu/elementwise/compare/where/JWhere.java
package com.wsr.gpu.elementwise.compare.where;

public class JWhere {
    public native void whereD0ToD0(long condition, float x, float y, long result, long runtime);
    public native void whereD0ToD1(long condition, float x, long y, long result, long runtime);
    public native void whereD1ToD0(long condition, long x, float y, long result, long runtime);
    public native void whereD1ToD1(long condition, long x, long y, long result, long runtime);
}

File: buffer/gpu/src/jvmAndroidMain/kotlin/com/wsr/knist/gpu/elementwise/compare/where/JWhere.kt
package com.wsr.knist.gpu.elementwise.compare.where

class JWhere {
    external fun whereD0ToD0(condition: Long, x: Float, y: Float, result: Long, runtime: Long)
    external fun whereD0ToD1(condition: Long, x: Float, y: Long, result: Long, runtime: Long)
    external fun whereD1ToD0(condition: Long, x: Long, y: Float, result: Long, runtime: Long)
    external fun whereD1ToD1(condition: Long, x: Long, y: Long, result: Long, runtime: Long)
}
